packages/app-builder-lib/src/targets/snap.ts

```diff
@@ -211,6 +211,11 @@ export default class SnapTarget extends Target {
     if (snap.compression != null) {
       args.push("--compression", snap.compression)
     }
+    
+    const publishConfig = getAppUpdatePublishConfiguration(packager, arch, false /* in any case validation will be done on publish */)
+    if (publishConfig != null) {
+      await outputFile(path.join(packager.getResourcesDir(stageDir.dir), "app-update.yml"), serializeToYaml(publishConfig))
+    }
```

Comment:
I don't understand how this change, being applied to only snap, resovles the issue for deb/rpm builds that also do not support auto-updates.